### ISSUE TITLE
[v2.0.0-alpha.19] Navigating to screen from modal crashes app

### ISSUE THREAD

**Issue by duhduhdan** [react-navigation/stack#248]
In `2.0.0-alpha.19`, I'm no longer able to navigate to a screen from a modal when the screen is not the first screen in the stack. It's possible I've been handling modals within a tab full of stacks incorrectly this whole time, but the desired functionality has worked until this latest release.

Specifically, the error is thrown when trying to evaluate `route.key`. In `2.0.0-alpha.19`, invoking `StackView`'s `handleTransitionComplete` in `handleCloseRoute` has changed from how it was working in `2.0.0-alpha.18`.

I whipped up a [very minimal example](https://github.com/duhduhdan/react-navigation/blob/b/route-key/example/src/TabStack.tsx) of what my configuration looks like today in production.

I also took a short recording to help visualize it more:
![nav_modal_nested_route](https://user-images.githubusercontent.com/4613917/66164240-12600700-e5e7-11e9-9ba8-0339a57a4bd4.gif)

If there's any more information I can provide, let me know.

**Comment by osdnk** [comment]
oh no

**Comment by osdnk** [comment]
Hey, @duhduhdan 
I pasted your code and it works. I only changed `Button` and `Text` to these from RN. 
https://streamable.com/jueo3

**Comment by duhduhdan** [comment]
@osdnk 

Thanks for taking a look, much appreciated. I ended up pasting the code into the playground for `react-navigation-stack` and saw the same results as the video you shared. Mind running the example in the link I provided? It's a fork from the `react-navigation` playground, so there shouldn't be too much of a headache spinning it up.

After testing it for a bit, I started wondering if the problem could be related to different versions of the pieces that make up react navigation. Ultimately, I made sure all of the core `react-navigation` dependencies between the two playgrounds were the same, but still ran into the problem on `react-navigation` playground and my production version. Not entirely sure what's going on here, but I'll keep playing around with the versions of _all_ the dependencies and see if I can gain anymore insight.

Thanks again for the great work on this library!
